librad/src/git/repo.rs

```diff
@@ -160,13 +166,21 @@ impl Repo {
         })?;
 
         Ok(Self {
-            id: urn.id.clone(),
+            urn: RadUrn {
+                path: uri::Path::empty(),
+                ..urn
+            },
             key,
             repo,
         })
     }
 
-    pub fn clone<T>(paths: &Paths, key: device::Key, url: &RadUrl) -> Result<Self, Error>
+    /// Clone a [`Repo`] from the given [`RadUrl`]
+    ///
+    /// The repo must not already exist.
+    ///
+    /// TODO: entity verification is not currently functional, pending #95
+    pub fn clone<T>(paths: &Paths, key: device::Key, url: RadUrl) -> Result<Self, Error>
```

Comment:
Ya, `clone()` is a trait method, whereas this `clone` is an inherent -- so it's not actually overloaded. 

I'm also not keen on using anything but git terminology here, because, you know, the module is called `git::repo`. `replicate` would also not be totally accurate, it's more like `like_git_clone_but_with_some_special_rules_see_the_spec_draft1_rev69_section_42`.

Reply:
Fair enough :smile: